File: client/bindings/python/examples/11_build_output.py
from iota_client import *
import json

# Create an IotaClient instance
client = IotaClient(nodes = ['https://api.testnet.shimmer.network'])

hex_address = client.bech32_to_hex("rms1qpllaj0pyveqfkwxmnngz2c488hfdtmfrj3wfkgxtk4gtyrax0jaxzt70zy")

address_unlock_condition = UnlockCondition(
    UnlockConditionType.Address,
    Address(AddressType.ED25519, hex_address)
)

# Build most basic output with amound and a single address unlock condition
basic_output = client.build_basic_output(
    unlock_conditions=[
        address_unlock_condition
    ],
    amount=1000000,
)
print(json.dumps(basic_output, indent=2))

# Output with metadata feature block
basic_output = client.build_basic_output(
    unlock_conditions=[
        address_unlock_condition,
    ],
    features=[
        Feature(FeatureType.Metadata, data='0x48656c6c6f2c20576f726c6421')
    ],
    amount=1000000,
)
print(json.dumps(basic_output, indent=2))

# Output with storage deposit return
basic_output = client.build_basic_output(
    unlock_conditions=[
        address_unlock_condition,
        UnlockCondition(type=UnlockConditionType.StorageDepositReturn, amount=1000000,
                        return_address=Address(AddressType.ED25519, hex_address))
    ],
    amount=1000000,
)
print(json.dumps(basic_output, indent=2))

# Output with expiration
basic_output = client.build_basic_output(
    unlock_conditions=[
        address_unlock_condition,
        UnlockCondition(type=UnlockConditionType.Expiration, unix_time=1,
                        return_address=Address(AddressType.ED25519, hex_address))
    ],
    amount=1000000,
)
print(json.dumps(basic_output, indent=2))

# Output with timelock
basic_output = client.build_basic_output(
    unlock_conditions=[
        address_unlock_condition,
        UnlockCondition(type=UnlockConditionType.Timelock, unix_time=1)
    ],
    amount=1000000,
)
print(json.dumps(basic_output, indent=2))

File: client/bindings/nodejs/examples/11_build_output.ts
// Copyright 2021-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Client, initLogger } from '@iota/client';
import type { UnlockConditionTypes } from '@iota/types';
require('dotenv').config({ path: '../.env' });

// Run with command:
// node ./dist/11_build_output.js

// Build a basic output
async function run() {
    initLogger();
    if (!process.env.NODE_URL) {
        throw new Error('.env NODE_URL is undefined, see .env.example');
    }

    const client = new Client({
        nodes: [process.env.NODE_URL],
    });

    try {
        const hexAddress = await client.bech32ToHex(
            'rms1qpllaj0pyveqfkwxmnngz2c488hfdtmfrj3wfkgxtk4gtyrax0jaxzt70zy',
        );

        const addressUnlockCondition: UnlockConditionTypes = {
            type: 0,
            address: {
                type: 0,
                pubKeyHash: hexAddress,
            },
        };

        // Build most basic output with amound and a single address unlock condition
        const basicOutput = await client.buildBasicOutput({
            amount: '1000000',
            unlockConditions: [addressUnlockCondition],
        });

        console.log(JSON.stringify(basicOutput, null, 2));

        // Output with metadata feature block
        const basicOutputWithMetadata = await client.buildBasicOutput({
            amount: '1000000',
            unlockConditions: [addressUnlockCondition],
            features: [
                {
                    type: 2,
                    // "Hello, World!" hex encoded
                    data: '0x48656c6c6f2c20576f726c6421',
                },
            ],
        });

        console.log(JSON.stringify(basicOutputWithMetadata, null, 2));

        // Output with storage deposit return
        const basicOutputWithStorageReturn = await client.buildBasicOutput({
            amount: '1000000',
            unlockConditions: [
                addressUnlockCondition,
                {
                    type: 1,
                    returnAddress: {
                        type: 0,
                        pubKeyHash: hexAddress,
                    },
                    amount: '1000000',
                },
            ],
        });

        console.log(JSON.stringify(basicOutputWithStorageReturn, null, 2));

        // Output with expiration
        const basicOutputWithExpiration = await client.buildBasicOutput({
            amount: '1000000',
            unlockConditions: [
                addressUnlockCondition,
                {
                    type: 3,
                    returnAddress: {
                        type: 0,
                        pubKeyHash: hexAddress,
                    },
                    unixTime: 1,
                },
            ],
        });

        console.log(JSON.stringify(basicOutputWithExpiration, null, 2));

        // Output with timelock
        const basicOutputWithTimelock = await client.buildBasicOutput({
            amount: '1000000',
            unlockConditions: [
                addressUnlockCondition,
                {
                    type: 2,
                    unixTime: 1,
                },
            ],
        });

        console.log(JSON.stringify(basicOutputWithTimelock, null, 2));
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
